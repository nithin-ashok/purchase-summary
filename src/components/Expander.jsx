import React from "react";

class Expander extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            expanded: false
        }

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(e) {
        e.preventDefault();
        const {expanded} = this.state;
        this.setState({
            expanded: !expanded
        });
    }


    render() {
        const {expandedTitle, collapsedTitle, children} = this.props;
        const {expanded} = this.state;
        const title = expanded ? collapsedTitle : expandedTitle;
        const glyph_icon = expanded ? "glyph-minus" : "glyph-plus";

        /* If there's no children, we don't need expander component at all*/
        if(!children) {
            return null
        }

        return (
            <div>
                <span className={`link ${glyph_icon}`} onClick={this.handleClick}>
                    {title}
                </span>
                {expanded && <div className="padded">
                    {children}
                </div>}

            </div>
        )
    }
}

export default Expander;