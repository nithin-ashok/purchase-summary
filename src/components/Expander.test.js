import React from 'react';
import Expander from "./Expander";
import renderer from 'react-test-renderer';

it("renderes correctly", () => {
    const tree = renderer.
        create(<Expander><span>Item Details</span></Expander>).toJSON();

    expect(tree).toMatchSnapshot();
});