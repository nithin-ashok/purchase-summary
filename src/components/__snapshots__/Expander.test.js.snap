// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes correctly 1`] = `
<div>
  <span
    className="link glyph-plus"
    onClick={[Function]}
  />
</div>
`;
